.github/workflows/benchmark_long.yml

```
@@ -0,0 +1,145 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Executes long-running benchmarks that are expected to take multple hours.
+
+name: Benchmark Long
+
+on:
+  schedule:
+    # Scheduled to run at 09:00 UTC and 21:00 UTC.
+    - cron: '0 09,21 * * *'
+  workflow_dispatch:
+  # TODO: Remove before submitting.
+  pull_request:
+
+env:
+  # This needs to be in env instead of the outputs of setup because it contains
+  # the run attempt and we want that to be the current attempt, not whatever
+  # attempt the setup step last ran in.
+  GCS_DIR: gs://iree-github-actions-${{ github.event_name == 'pull_request' && 'presubmit' || 'postsubmit' }}-artifacts/${{ github.run_id }}/${{ github.run_attempt }}
+
+# Jobs are organized into groups and topologically sorted by dependencies
+jobs:
+  setup:
+    uses: ./.github/workflows/setup.yml
+
+  benchmark_setup:
+    needs: setup
+    if: fromJson(needs.setup.outputs.should-run)
+    runs-on: ubuntu-20.04
+    env:
+      BENCHMARK_PRESETS: cuda-long
+    outputs:
+      benchmark-presets: ${{ steps.configure.outputs.benchmark-presets }}
+    steps:
+      - name: "Setting benchmark presets"
+        id: configure
+        run: |
+          echo "benchmark-presets=${BENCHMARK_PRESETS}" >> $GITHUB_OUTPUT
```

Do we need a step for the benchmark presets? Seems like we just set a constant for it. Maybe define a workflow level env variable and use it below?

```suggestion
env:
    BENCHMARK_PRESETS: cuda-long
```

---

Created an env var and removed the step.

---

This doesn't seem to work. Using an environment variable as a parameter to a re-usable workflow gives an error. It looks like it needs to use an output to a previous job or step: https://github.com/orgs/community/discussions/26671

---

Yeah this is why we have GCS_DIR repeated everywhere. This is a really annoying GitHub limitation. It can just be a literal though and it's only used in one place, so I would just do that.

---

Made it a literal and removed the step.